src/codeocean/data_asset.py

```diff
@@ -331,3 +331,16 @@ def transfer_data_asset(self, data_asset_id: str, transfer_params: TransferDataP
             f"data_assets/{data_asset_id}/transfer",
             json=transfer_params.to_dict()
         )
+
+    def paginate_search(self, data_asset_search_params: DataAssetSearchParams) -> Iterator[DataAsset]:
+        has_more = True
+        search_params = data_asset_search_params.to_dict()
+        while has_more:
+            response = self.search_data_assets(
+                search_params=DataAssetSearchParams(**search_params)
+            )
+            next_token = response.next_token
+            has_more = response.has_more
+            search_params["next_token"] = next_token
+            for result in response.results:
+                yield result
```

Comment:
```suggestion
    def search_data_assets_iterator(self, search_params: DataAssetSearchParams) -> Iterator[DataAsset]:
        params = search_params.to_dict()
        while True:
            response = self.search_data_assets(
                search_params=DataAssetSearchParams(**params)
            )

            for result in response.results:
                yield result
                
            if not response.has_more:
                return
                
            params["next_token"] = response.next_token
```

I'm suggesting a method name change as this is iterating over results